src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,27 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        try (Session session = factory.openSession()) {
+            Transaction transaction = session.beginTransaction();
+            try {
+                session.persist(actor);
+                transaction.commit();
+            } catch (Exception e) {
+                if (transaction != null) {
+                    transaction.rollback();
+                }
+                throw new DataProcessingException(
+                        "Can not to add actor with name: " + actor.getName());
+            }
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        Session session = factory.openSession();
+        Actor actor = session.get(Actor.class, id);
+        session.close();
+        return Optional.ofNullable(actor);
     }
```

Comment:
Please use try with resources here. Also, you can use this: Optional.ofNullable(session.get(Actor.class, id))